### PR TITLE
feat: handle collection unverification for reg NFTs

### PR DESCRIPTION
**Pull request by tahsintunan**
## Overview
-   Handle collection unverification by setting `group_value` in the `asset_grouping` table to `null`. 
- This required a schema migration to remove the `NOT NULL` constraint from the `group_value` column in the `asset_grouping` table.

## Testing
-   Testing performed locally to validate the changes

### CODE REVIEW

File: nft_ingester/src/program_transformers/token_metadata/v1_asset.rs [hunk 1]
```diff
@@ -297,6 +299,35 @@ pub async fn save_v1_asset<T: ConnectionTrait + TransactionTrait>(
...
+                    ])
+                    .update_columns([
+                        asset_grouping::Column::GroupValue,
+                        asset_grouping::Column::SlotUpdated,
+                    ])
+                    .to_owned(),
+                )
+                .build(DbBackend::Postgres);
+            query.sql = format!(
+                "{} WHERE excluded.slot_updated > asset_grouping.slot_updated AND excluded.seq >= asset_grouping.seq",
```
Comment (tahsintunan) [inline nft_ingester/src/program_transformers/token_metadata/v1_asset.rs:325]: Here I've wrote `AND excluded.seq >= asset_grouping.seq` to be consistent with previous ones. But this is not required.

File: migration/src/m20230726_013107_remove_not_null_constraint_from_group_value.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                sea_query::Table::alter()
+                    .table(asset_grouping::Entity)
+                    .modify_column(ColumnDef::new(asset_grouping::Column::GroupValue).null())
+                    .to_owned(),
+            )
+            .await?;
+        Ok(())
+    }
+
+    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
```
Comment (tahsintunan) [inline migration/src/m20230726_013107_remove_not_null_constraint_from_group_value.rs:21]: _**Something to consider:**_
If the column already has `NULL` values, the rollback won't work. For that, we'll need to remove/modify the `NULL` values first. Any suggestions here would be helpful.

Comment (NicolasPennie) [reply migration/src/m20230726_013107_remove_not_null_constraint_from_group_value.rs:21]: Add a comment explaining this and that's ok

File: nft_ingester/src/program_transformers/token_metadata/v1_asset.rs [hunk 3]
```diff
@@ -297,6 +299,35 @@ pub async fn save_v1_asset<T: ConnectionTrait + TransactionTrait>(
             txn.execute(query)
                 .await
                 .map_err(|db_err| IngesterError::AssetIndexError(db_err.to_string()))?;
+        } else {
+            let model = asset_grouping::ActiveModel {
+                asset_id: Set(id.to_vec()),
+                group_key: Set("collection".to_string()),
+                group_value: Set(None),
```
[the remark this answers is not in the document]
Comment (NicolasPennie) [reply nft_ingester/src/program_transformers/token_metadata/v1_asset.rs:306]: `group_value = verify ? collection : None`

File: digital_asset_types/src/dao/scopes/asset.rs [hunk 4]
```diff
@@ -290,6 +290,7 @@ pub async fn get_by_id(
         .await?;
     let grouping: Vec<asset_grouping::Model> = asset_grouping::Entity::find()
         .filter(asset_grouping::Column::AssetId.eq(asset.id.clone()))
+        .filter(asset_grouping::Column::GroupValue.is_not_null())
```
Comment (NicolasPennie) [inline digital_asset_types/src/dao/scopes/asset.rs:293]: Need to add filter here: https://github.com/helius-labs/digital-asset-rpc-infrastructure/blob/6a2081dde73e760ca29c5c0bb90160e0e3e54464/digital_asset_types/src/dao/scopes/asset.rs#L291

**Review by NicolasPennie — APPROVED** [review]

**Pull request merged**